volume/store/store.go

```diff
@@ -141,7 +141,7 @@ func (s *VolumeStore) Purge(name string) {
 // VolumeStore is a struct that stores the list of volumes available and keeps track of their usage counts
 type VolumeStore struct {
 	locks      *locker.Locker
-	globalLock sync.Mutex
+	globalLock sync.RWMutex
```

Comment:
+1

```
// globalLock is used to protect access to mutable structures used by the store object
```

```
// locks ensures that only one action is being performed on a particular volume at a time without locking the entire store
// since actions on volumes can be quite slow, this ensures the store is free to handle requests for other volumes.
```